### PR TITLE
Make Timeout Configureable

### LINKED ISSUE #36: Make timeout dynamically configurable
Currently the [GraphHopperWeb implementation](https://github.com/graphhopper/directions-api-java-client/blob/master/client/src/main/java/com/graphhopper/api/GraphHopperWeb.java#L71) has a fixed timeout of 5 seconds.

5 seconds are fine when executing a CH routing request. However, a non ch routing request might take longer. Or if having a bad internet connection one might want to increase the timeout as well.

Therefore I think, it would be great to be able change the timeout on a per method basis. Currently, this would be possible with[ setting a new instance](https://github.com/graphhopper/directions-api-java-client/blob/master/client/src/main/java/com/graphhopper/api/GraphHopperWeb.java#L96) of the OKHttpClient before the request.

WDYT?

### PR DESCRIPTION
**Pull request by boldtrn**
This PR fixes #36. With this PR we are now able to dynamically set the timeout of the client on a per method base.

Let me know what you think and if I should change anything.

### CODE REVIEW

File: client/src/test/java/com/graphhopper/api/GraphHopperWebIT.java [hunk 1]
```diff
@@ -55,6 +56,25 @@ public void testSimpleRoute() {
...
 
     @Test
+    public void testTimeout() {
+        GHRequest req = new GHRequest().
+                addPoint(new GHPoint(49.6724, 11.3494)).
+                addPoint(new GHPoint(49.6550, 11.4180));
+        GHResponse res = gh.route(req);
+        assertFalse("errors:" + res.getErrors().toString(), res.hasErrors());
+
+        req.getHints().put(GraphHopperWeb.TIMEOUT, 1);
```
Comment (karussell) [inline client/src/test/java/com/graphhopper/api/GraphHopperWebIT.java]: What are you testing here?

Comment (boldtrn) [inline client/src/test/java/com/graphhopper/api/GraphHopperWebIT.java]: Setting the timeout very small to get a timeout exception. Not sure if this makes sense? Another possibility could be to make the method `getClientForRequest` so I could check the timeout value of the client.

Comment (karussell) [inline client/src/test/java/com/graphhopper/api/GraphHopperWebIT.java]: Ah, no, good idea. Then probably put an assert in the exception block and the fail directly after the method call?

Comment (boldtrn) [inline client/src/test/java/com/graphhopper/api/GraphHopperWebIT.java]: Thanks for pointing that out, I changed the test.

**Comment by karussell** [comment]
BTW: why this?

> Failed tests:   geocodeGetTestNominatim(io.swagger.client.api.GeocodingApiTest): expected:<null> but was:<de>

**Comment by boldtrn** [comment]
> BTW: why this?

Thanks, haven't seen that. See #41

**Pull request merged**